---
title: "S4 Hatchery Microbiome Analysis"
author: "Evelyn Takyi, updated by Rebecca Stevick"
date: "`r Sys.Date()`"
output:
  html_document:
    toc: yes
    keep_md: yes
    theme: cerulean
    toc_float:
      collapsed: no
      smooth_scroll: no
  pdf_document:
    toc: yes
---

# Setup

## Load packages

```{r packages}

library(readxl)
library(openxlsx)
library(vegan)
library(tidyverse)
library(phyloseq)
library(gridExtra)
library(grid)
library(ggpubr)
library(agricolae)
library(microbiome)
library(devtools)
library(ggh4x) # devtools::install_github("teunbrand/ggh4x")
library(patchwork)
library(lme4)
library(car)
library(emmeans)

set.seed(121)

# set global options for code output
knitr::opts_chunk$set(echo=TRUE, warning=FALSE)


```

## Load the data

```{r load}

## Count table
abund_table <- read.xlsx("Data.xlsx", sheet="table", rowNames = TRUE)

## Metadata table
meta_table <-read.xlsx("Data.xlsx", sheet="meta_table", rowNames = TRUE)

## Taxonomy table
taxa_table <- read_excel("Data.xlsx", sheet = "taxonomy") %>% 
   separate(Taxon, sep="; ", c("Kingdom", "Phylum", "Class", "Order", "Family", "Genus", "Species")) %>% 
   select(-c(Confidence)) %>% 
   column_to_rownames("FeatureID")
taxa_table[is.na(taxa_table)] <- 'Unknown'

```

## Format data as physeq

```{r format}

## abundance table
count_tab_phy <- otu_table(as.matrix(abund_table), taxa_are_rows=T)

# taxonomy 
tax_tab_phy <- tax_table(as.matrix(taxa_table))

## metadata
sample_info_tab_phy <- sample_data(meta_table)

## make phyloseq object
All_S4 <-phyloseq(count_tab_phy, tax_tab_phy, sample_info_tab_phy)

##Check if there are ASV's with 0 counts, number and remove them
All_S4_prune <- prune_taxa(taxa_sums(All_S4) > 0, All_S4)
any(taxa_sums(All_S4_prune) == 0)

##Remove any taxa associated with chloroplast and mitochondria
justbacteria <- All_S4_prune %>%
  subset_taxa(Kingdom == "d__Bacteria" & Phylum  != "p__Cyanobacteria" & Family != "f__Mitochondria")


```

## Reorder metadata factors

```{r reorder}

# Reorder Trials
sample_data(justbacteria)$Trial <- factor(sample_data(justbacteria)$Trial, levels = c("T1","T2","T3","T4","T5","T6","T7","T8"))
levels(sample_data(justbacteria)$Trial)

# Relabel the Trial
sample_data(justbacteria)$Name <- factor(sample_data(justbacteria)$Name, labels = c("T1","T2","T3","T4","T5","T6","T7","T8_NUV","T8_UV"))
levels(sample_data(justbacteria)$Name)

# Reorder Hatchery labels
sample_data(justbacteria)$Hatchery <- factor(sample_data(justbacteria)$Hatchery, levels = c("VIMS", "Mat", "MK", "RWU"))
levels(sample_data(justbacteria)$Hatchery)

```


## Rarefy the data

```{r rarefy}

#Delete samples less than 12000 reads
justbacteria_12000 <- subset_samples(justbacteria, sample_sums(justbacteria) > 12000)

# check size of smallest sample
min(sample_sums(justbacteria_12000))

# Rarefy to 12640 reads
justbacteria_12000_rarefy <- rarefy_even_depth(justbacteria, rngseed= 81, sample.size = min(sample_sums(justbacteria_12000)))

# check the total number of ASVs
length(taxa_sums(justbacteria_12000_rarefy))

## Rearrange the Metatable
SD <- sample_data(justbacteria_12000_rarefy) %>%
    data.frame() %>%
    select("Treatment","Trial", "Num","Tank_Replicate","Hatchery","Location", "Condition", "Num") %>% 
    rownames_to_column("SampleID")

# remove technical replicates from meta table
All_Meta <- SD %>% distinct(Num,Treatment,Trial,Hatchery, Location, Tank_Replicate, Condition) %>% 
   remove_rownames %>% column_to_rownames(var="Num")

```


# Figure 1 - Order barplot

```{r, orderbars, fig.width=17, fig.height=10}

my_colors <- c("#CBD588", "#5F7FC7", "orange","#DA5724", "#508578", "#CD9BCD", "#AD6F3B", "#673770","#D14285", "#652926", "#C84248", "#8569D5", "grey60","#D1A33D")

Order_abundances <- justbacteria_12000_rarefy %>% tax_glom(taxrank = "Order") %>% 
   psmelt() %>% 
   # calculate mean of the technical replicates per order
   group_by(Num,Treatment,Name,Hatchery,Order) %>% 
   summarise(mean_abun=mean(Abundance/12640*100)) %>% ungroup() %>% 
   # make new column of cleaned top 12 Orders
   mutate(OrderClean=str_remove(Order, "o__"),
          # select the 12 most abundant orders, then group all the Others
          OrderOther=forcats::fct_lump_n(f=OrderClean, w=mean_abun, other_level="Others", n=12)) %>% 
   # calculate sums for Other
   group_by(Num,Treatment,Name,Hatchery,OrderOther) %>% 
   summarise(sum_abun=sum(mean_abun)) %>% ungroup() %>% 
   # reorder OrderOther
   mutate(OrderOther = reorder(OrderOther, -sum_abun),
          OrderOther = fct_relevel(OrderOther, "Others", after = Inf))
  Order_abundances
  write.csv( Order_abundances, "Tables/order.csv")
   
   # start plotting
   Order_abundances %>% ggplot() +
   geom_col(aes(x = Num, y = sum_abun, fill = OrderOther), position="fill")+
   labs(y="Percent Order Abundance", x=NULL, fill="Order") +
   facet_nested(.~Hatchery+Name+Treatment, scales = "free", space="free", switch="x") +
   scale_fill_manual(values = my_colors) +
   scale_y_continuous(expand = c(0,0),labels=scales::percent)+
   theme_minimal()+
   theme(axis.text.x = element_blank(), panel.spacing = unit(0.2,"lines"),
         panel.grid.major.x = element_blank(), panel.grid.minor.x = element_blank(),
         strip.background = element_rect(fill="gray85"),
         axis.text.y.left = element_text(size=16, face="bold"),
         axis.title.y = element_text(size = 18, face="bold"),
         strip.text.x = element_text(size = 18,face="bold"),
         legend.position="bottom", legend.title = element_text(size=20, face="bold"), 
         legend.text = element_text(size=18, face="bold"))

ggsave("Figures/Figure1.png", width=18, height=8, dpi=400)

```



# Figure 2 - Simpsons diversity

```{r, simpsonsall}

# subset samples to remove Trial 5 and trial6
phy <- subset_samples(justbacteria_12000_rarefy, Trial %in% c("T1","T2","T3","T4","T7","T8"))

# get metadata
SD <- sample_data(phy) %>% data.frame() %>%
   select("Treatment","Trial","Num","Tank_Replicate") %>% 
   rownames_to_column("SampleID")

# calculate diversity
adiv <- data.frame(
   "Simpson" = phyloseq::estimate_richness(phy, measures = "Simpson"),
   "Chao1"= phyloseq::estimate_richness(phy, measures = "Chao1"),
   "Status" = phyloseq::sample_data(phy)$Treatment) %>% 
   rownames_to_column("SampleID") %>% 
   # add in the metadata
   left_join(SD)

# calculate diversity per tank from technical replicates
Div <- adiv %>% group_by(Num,Trial,Treatment,Tank_Replicate) %>% summarise(mean_div=mean(Simpson), sd_div=sd(Simpson))

Div <- as.data.frame(Div)
Div

# plotting
Div_SIM <- ggplot(Div, aes(x=Treatment, y=mean_div, fill=Treatment))+
   geom_boxplot(alpha=0.7)+
   geom_jitter(aes(shape=Treatment), width=0.2)+
   scale_fill_manual(values=c("blue", "orange"))+
   scale_shape_manual(values=c(22,21))+
   facet_grid(.~Trial, scales="free",space="free_x",switch="y")+
   theme_gray()+
   labs(y="Simpson's Diversity", x="Treatment", fill="Treatment")+theme_bw()+
   theme(strip.text.x = element_text(size=12,face="bold"),
         panel.grid.major.x = element_blank(),
         strip.background = element_rect(fill="gray85"),
         axis.text.y = element_text(size="14", color="black"), 
         axis.title.y = element_text(face="bold",size="14", color="black"),
         axis.text.x = element_text(size="12", color="black", angle = 0), 
         axis.title.x.bottom = element_text(face="bold",size="14", color="black"),
         axis.text.x.top= element_text(face="bold",size="14", color="black"),
         legend.position = "None",panel.spacing = unit(0.2, "mm"),               
         panel.border = element_rect(colour="grey"))+
   scale_y_continuous(limits=c(0.6, 1))
Div_SIM 

ggsave(filename = "Figures/Figure2.png", width=7, height=4, dpi = 400)
```

## Stats

```{r}

Avg_Simp <- adiv %>% group_by(Num,Trial,Treatment,Tank_Replicate) %>% summarise(mean_Simpsons=mean(Simpson), sd_Simpsons=sd(Simpson))

Avg_Chao1 <- adiv %>% group_by(Num,Trial,Treatment,Tank_Replicate) %>% summarise(mean_Chao1=mean(Chao1.Chao1), sd_Chao1=sd(Chao1.Chao1))

Average <- full_join(Avg_Chao1, Avg_Simp)
write.csv(Average, "Tables/diversity.csv")
##Simpson's diversity
Simpson_Anova <- aov(mean_Simpsons ~ Treatment*Trial, data=Average)
summary(Simpson_Anova)
tukey_Simp <- HSD.test(Simpson_Anova, "Trial", group = FALSE)
tukey_Simp$comparison

##Chao1 richness
Chao1_Anova <- aov(mean_Chao1 ~ Treatment*Trial, data=Average)
summary(Chao1_Anova)
tukey_Chao1 <- HSD.test(Chao1_Anova, "Trial", group = FALSE)
tukey_Chao1$comparison

```

# Figure 3 - NMDS plots

```{r betasetup}

#Generate ellipse points 
veganCovEllipse<-function (cov, center = c(0, 0), scale = 1, npoints = 100) 
{
  theta <- (0:npoints) * 2 * pi/npoints
  Circle <- cbind(cos(theta), sin(theta))
  t(center + scale * t(Circle %*% chol(cov)))
}

```

## Trial 1

```{r beta1}

ASV_t1 <- subset_samples(justbacteria_12000_rarefy, Trial=="T1") %>% psmelt() %>%
   group_by(Num,OTU) %>% summarise(avg_abundance = mean(Abundance)) %>%  # average by tank
   spread(Num,avg_abundance)  %>% remove_rownames %>% column_to_rownames(var="OTU") %>%  t() %>% as.data.frame()
Meta_t1 <- All_Meta %>% filter(Trial=="T1")

sol_t1 <-metaMDS(ASV_t1 ,distance = "bray", k = 2, trymax = 50)
NMDS_t1=data.frame(NMDS1=sol_t1$point[,1], NMDS2=sol_t1$point[,2], Treatment=as.factor(Meta_t1[,1]))

plot.new()
ord_t1<-ordiellipse(sol_t1, as.factor(Meta_t1$Treatment), display = "sites", kind ="sd", conf = 0.95, label = T)
#dev.off()

df_ell_t1 <- data.frame()
for(g in levels(NMDS_t1$Treatment)){
   if(g!="" && (g %in% names(ord_t1))){
      df_ell_t1 <- rbind(df_ell_t1, cbind(as.data.frame(with(NMDS_t1[NMDS_t1$Treatment==g,],                            veganCovEllipse(ord_t1[[g]]$cov,ord_t1[[g]]$center,ord_t1[[g]]$scale))),Treatment=g))}}

#Calculate permanova p-value:
adon_t1<-adonis2(ASV_t1 ~Treatment, data=Meta_t1, by=NULL,method="bray", k=2)
adon_t1
# calculate beta-dispersion p-value:
disper_t1 <- betadisper(vegdist(ASV_t1, distance="bray"), Meta_t1$Treatment)
anova(disper_t1)


#Plot NMDS
NMDSplot_t1<-ggplot(data=NMDS_t1,aes(NMDS1,NMDS2,col=Treatment))+
   annotate("text",x=min(NMDS_t1$NMDS1-0.2),y=min(NMDS_t1$NMDS2-0.22), 
            label=paste("p-value = ", round(adon_t1$`Pr(>F)`[1],2)), hjust=0)+
   annotate("text",x=min(NMDS_t1$NMDS1-0.2),y=min(NMDS_t1$NMDS2-0.15), 
            label=paste("Stress = ", round(sol_t1$stress,2)), hjust=0)+
   geom_path(data=df_ell_t1, aes(x=NMDS1, y=NMDS2, linetype=Treatment), size=1)+
   scale_linetype_manual(values=c("C"="solid","S4"="dotted"))+
   scale_colour_manual(values=c("C"="blue","S4"="orange"))+
   ggtitle("Trial 1, VIMS")+
   geom_point(aes(shape=Treatment), size=4, alpha=0.8) + 
   scale_shape_manual(values=c(15,16)) + 
   theme_grey()+
   theme(plot.title=element_text(size = 16, face = "bold", hjust = 0.5))

NMDSplot_t1

```


## Trial 2

```{r beta2}

ASV_t2 <- subset_samples(justbacteria_12000_rarefy, Trial=="T2") %>% psmelt() %>%
   group_by(Num,OTU) %>% summarise(avg_abundance = mean(Abundance)) %>%  # average by tank
   spread(Num,avg_abundance)  %>% remove_rownames %>% column_to_rownames(var="OTU") %>%  t() %>% as.data.frame()
Meta_t2 <- All_Meta %>% filter(Trial=="T2")

sol_t2 <-metaMDS(ASV_t2 ,distance = "bray", k = 2, trymax = 50)
NMDS_t2=data.frame(NMDS1=sol_t2$point[,1], NMDS2=sol_t2$point[,2], Treatment=as.factor(Meta_t2[,1]))

plot.new()
ord_t2<-ordiellipse(sol_t2, as.factor(Meta_t2$Treatment), display = "sites", kind ="sd", conf = 0.95, label = T)
#dev.off()

df_ell_t2 <- data.frame()
for(g in levels(NMDS_t2$Treatment)){
   if(g!="" && (g %in% names(ord_t2))){
      df_ell_t2 <- rbind(df_ell_t2, cbind(as.data.frame(with(NMDS_t2[NMDS_t2$Treatment==g,],                            veganCovEllipse(ord_t2[[g]]$cov,ord_t2[[g]]$center,ord_t2[[g]]$scale))),Treatment=g))}}

#Calculate p-value:
adon_t2<-adonis2(ASV_t2 ~Treatment, data=Meta_t2, by=NULL,method="bray", k=2)
adon_t2
# calculate beta-dispersion p-value:
disper_t2 <- betadisper(vegdist(ASV_t2, distance="bray"), Meta_t2$Treatment)
anova(disper_t2)

#Plot NMDS
NMDSplot_t2<-ggplot(data=NMDS_t2,aes(NMDS1,NMDS2,col=Treatment))+
   annotate("text",x=min(NMDS_t2$NMDS1-0.2),y=min(NMDS_t2$NMDS2-0.2), 
            label=paste("p-value = ", round(adon_t2$`Pr(>F)`[1],3)), hjust=0)+
   annotate("text",x=min(NMDS_t2$NMDS1-0.2),y=min(NMDS_t2$NMDS2-0.15), 
            label=paste("Stress = ", round(sol_t2$stress,3)), hjust=0)+
   geom_path(data=df_ell_t2, aes(x=NMDS1, y=NMDS2, linetype=Treatment), size=1)+
   scale_linetype_manual(values=c("C"="solid","S4"="dotted"))+
   scale_colour_manual(values=c("C"="blue","S4"="orange"))+
   ggtitle("Trial 2, VIMS")+
   geom_point(aes(shape=Treatment), size=4, alpha=0.8) + 
   scale_shape_manual(values=c(15,16)) + 
   theme_grey()+
   theme(plot.title=element_text(size = 16, face = "bold", hjust = 0.5))


NMDSplot_t2

```


## Trial 3

```{r beta3}

ASV_t3 <- subset_samples(justbacteria_12000_rarefy, Trial=="T3") %>% psmelt() %>%
   group_by(Num,OTU) %>% summarise(avg_abundance = mean(Abundance)) %>%  # average by tank
   spread(Num, avg_abundance) %>% column_to_rownames(var="OTU") %>% t() %>% as.data.frame()
Meta_t3 <- All_Meta %>% filter(Trial=="T3")

sol_t3 <- metaMDS(ASV_t3, distance = "bray", k = 2)
NMDS_t3=data.frame(NMDS1=sol_t3$point[,1], NMDS2=sol_t3$point[,2], Treatment=as.factor(Meta_t3[,1]))

plot.new()
ord_t3<-ordiellipse(sol_t3, as.factor(Meta_t3$Treatment), display = "sites", kind ="sd", conf = 0.95, label = T)
#dev.off()

df_ell_t3 <- data.frame()
for(g in levels(NMDS_t3$Treatment)){
  if(g!="" && (g %in% names(ord_t3))){
    df_ell_t3 <- rbind(df_ell_t3, cbind(as.data.frame(with(NMDS_t3[NMDS_t3$Treatment==g,],                            veganCovEllipse(ord_t3[[g]]$cov,ord_t3[[g]]$center,ord_t3[[g]]$scale))),Treatment=g))}}

#Calculate p-value:
adon_t3<-adonis2(ASV_t3 ~Treatment, data=Meta_t3, by=NULL,method="bray", k=2)
adon_t3
# calculate beta-dispersion p-value:
disper_t3 <- betadisper(vegdist(ASV_t3, distance="bray"), Meta_t3$Treatment)
anova(disper_t3)

#Plot NMDS
NMDSplot_t3<-ggplot(data=NMDS_t3,aes(NMDS1,NMDS2,col=Treatment))+
  annotate("text",x=min(NMDS_t3$NMDS1-0.6),y=min(NMDS_t3$NMDS2-0.2), 
           label=paste("p-value = ", round(adon_t3$`Pr(>F)`[1],2)), hjust=0)+
  annotate("text",x=min(NMDS_t3$NMDS1-0.6),y=min(NMDS_t3$NMDS2-0.15), 
          label=paste("Stress = ", round(sol_t3$stress,2)), hjust=0)+
  geom_path(data=df_ell_t3, aes(x=NMDS1, y=NMDS2, linetype=Treatment), size=1)+
  scale_linetype_manual(values=c("C"="solid","S4"="dotted"))+
  scale_colour_manual(values=c("C"="blue","S4"="orange"))+
  ggtitle("Trial 3, VIMS")+
   geom_point(aes(shape=Treatment), size=4, alpha=0.8) + 
   scale_shape_manual(values=c(15,16)) + 
   theme_grey()+
   theme(plot.title=element_text(size = 16, face = "bold", hjust = 0.5))



NMDSplot_t3

```

## Trial 4

```{r beta4}

ASV_t4 <- subset_samples(justbacteria_12000_rarefy, Trial=="T4") %>% psmelt() %>%
   group_by(Num,OTU) %>% summarise(avg_abundance = mean(Abundance)) %>%  # average by tank
   spread(Num, avg_abundance) %>% column_to_rownames(var="OTU") %>% t() %>% as.data.frame()
Meta_t4 <- All_Meta %>% filter(Trial=="T4")

sol_t4 <- metaMDS(ASV_t4, distance = "bray", k = 2)
NMDS_t4=data.frame(NMDS1=sol_t4$point[,1], NMDS2=sol_t4$point[,2], Treatment=as.factor(Meta_t4[,1]))

plot.new()
ord_t4<-ordiellipse(sol_t4, as.factor(Meta_t4$Treatment), display = "sites", kind ="sd", conf = 0.95, label = T)
#dev.off()

df_ell_t4 <- data.frame()
for(g in levels(NMDS_t4$Treatment)){
  if(g!="" && (g %in% names(ord_t4))){
    df_ell_t4 <- rbind(df_ell_t4, cbind(as.data.frame(with(NMDS_t4[NMDS_t4$Treatment==g,],                            veganCovEllipse(ord_t4[[g]]$cov,ord_t4[[g]]$center,ord_t4[[g]]$scale))),Treatment=g))}}

#Calculate p-value:
adon_t4<-adonis2(ASV_t4 ~Treatment, data=Meta_t4, by=NULL,method="bray", k=2)
adon_t4
# calculate beta-dispersion p-value:
disper_t4 <- betadisper(vegdist(ASV_t4, distance="bray"), Meta_t4$Treatment)
anova(disper_t4)

#Plot NMDS
NMDSplot_t4<-ggplot(data=NMDS_t4,aes(NMDS1,NMDS2,col=Treatment))+
  annotate("text",x=min(NMDS_t4$NMDS1-0.35),y=min(NMDS_t4$NMDS2-0.2), 
           label=paste("p-value = ", round(adon_t4$`Pr(>F)`[1],2)), hjust=0)+
  annotate("text",x=min(NMDS_t4$NMDS1-0.35),y=min(NMDS_t4$NMDS2-0.15), 
           label=paste("Stress = ", round(sol_t4$stress,2)), hjust=0)+
  geom_path(data=df_ell_t4, aes(x=NMDS1, y=NMDS2, linetype=Treatment), size=1)+
  scale_linetype_manual(values=c("C"="solid","S4"="dotted"))+
  scale_colour_manual(values=c("C"="blue","S4"="orange"))+
  ggtitle("Trial 4, VIMS")+
   geom_point(aes(shape=Treatment), size=4, alpha=0.8) + 
   scale_shape_manual(values=c(15,16)) + 
   theme_grey()+
   theme(plot.title=element_text(size = 16, face = "bold", hjust = 0.5))


NMDSplot_t4

```


## Trial 7

```{r beta7}

ASV_t7 <- subset_samples(justbacteria_12000_rarefy, Trial=="T7") %>% psmelt() %>%
   group_by(Num,OTU) %>% summarise(avg_abundance = mean(Abundance)) %>%  # average by tank
   spread(Num, avg_abundance) %>% column_to_rownames(var="OTU") %>% t() %>% as.data.frame()
Meta_t7 <- All_Meta %>% filter(Trial=="T7")

sol_t7 <- metaMDS(ASV_t7, distance = "bray", k = 2)
NMDS_t7=data.frame(NMDS1=sol_t7$point[,1], NMDS2=sol_t7$point[,2], Treatment=as.factor(Meta_t7[,1]))

plot.new()
ord_t7<-ordiellipse(sol_t7, as.factor(Meta_t7$Treatment), display = "sites", kind ="sd", conf = 0.95, label = T)
#dev.off()

df_ell_t7 <- data.frame()
for(g in levels(NMDS_t7$Treatment)){
  if(g!="" && (g %in% names(ord_t7))){
    df_ell_t7 <- rbind(df_ell_t7, cbind(as.data.frame(with(NMDS_t7[NMDS_t7$Treatment==g,],                            veganCovEllipse(ord_t7[[g]]$cov,ord_t7[[g]]$center,ord_t7[[g]]$scale))),Treatment=g))}}

#Calculate p-value:
adon_t7<-adonis2(ASV_t7 ~Treatment, data=Meta_t7, by=NULL,method="bray", k=2)
adon_t7
# calculate beta-dispersion p-value:
disper_t7 <- betadisper(vegdist(ASV_t7, distance="bray"), Meta_t7$Treatment)
anova(disper_t7)

#Plot NMDS
NMDSplot_t7<-ggplot(data=NMDS_t7,aes(NMDS1,NMDS2,col=Treatment))+
  annotate("text",x=min(NMDS_t7$NMDS1-0.2),y=min(NMDS_t7$NMDS2-0.34), 
           label=paste("p-value = ", round(adon_t7$`Pr(>F)`[1],2)), hjust=0)+
  annotate("text",x=min(NMDS_t7$NMDS1-0.2),y=min(NMDS_t7$NMDS2-0.28), 
           label=paste("Stress = ", round(sol_t7$stress,2)), hjust=0)+
  geom_path(data=df_ell_t7, aes(x=NMDS1, y=NMDS2, linetype=Treatment), size=1)+
  scale_linetype_manual(values=c("C"="solid","S4"="dotted"))+
  scale_colour_manual(values=c("C"="blue","S4"="orange"))+
  ggtitle("Trial 7, MK")+
   geom_point(aes(shape=Treatment), size=4, alpha=0.8) + 
   scale_shape_manual(values=c(15,16)) + 
   theme_grey()+
   theme(plot.title=element_text(size = 16, face = "bold", hjust = 0.5))


NMDSplot_t7

```


## Trial 8

```{r beta8}

ASV_t8 <- subset_samples(justbacteria_12000_rarefy, Trial=="T8") %>% psmelt() %>%
   group_by(Num,OTU) %>% summarise(avg_abundance = mean(Abundance)) %>%  # average by tank
   spread(Num, avg_abundance) %>% column_to_rownames(var="OTU") %>% t() %>% as.data.frame()
Meta_t8 <- All_Meta %>% filter(Trial=="T8")

sol_t8 <- metaMDS(ASV_t8, distance = "bray", k = 2)
NMDS_t8=data.frame(NMDS1=sol_t8$point[,1], NMDS2=sol_t8$point[,2], Treatment=as.factor(Meta_t8[,1]))

plot.new()
ord_t8<-ordiellipse(sol_t8, as.factor(Meta_t8$Treatment), display = "sites", kind ="sd", conf = 0.95, label = T)
#dev.off()

df_ell_t8 <- data.frame()
for(g in levels(NMDS_t8$Treatment)){
  if(g!="" && (g %in% names(ord_t8))){
    df_ell_t8 <- rbind(df_ell_t8, cbind(as.data.frame(with(NMDS_t8[NMDS_t8$Treatment==g,],                            veganCovEllipse(ord_t8[[g]]$cov,ord_t8[[g]]$center,ord_t8[[g]]$scale))),Treatment=g))}}

#Calculate p-value:
adon_t8<-adonis2(ASV_t8 ~Treatment, data=Meta_t8, by=NULL,method="bray", k=2)
adon_t8
# calculate beta-dispersion p-value:
disper_t8 <- betadisper(vegdist(ASV_t8, distance="bray"), Meta_t8$Treatment)
anova(disper_t8)

#Plot NMDS
NMDSplot_t8<-ggplot(data=NMDS_t8,aes(NMDS1,NMDS2,col=Treatment))+
  annotate("text",x=min(NMDS_t8$NMDS1-0.5),y=min(NMDS_t8$NMDS2-0.28), 
           label=paste("p-value = ", round(adon_t8$`Pr(>F)`[1],2)), hjust=0)+
  annotate("text",x=min(NMDS_t8$NMDS1-0.5),y=min(NMDS_t8$NMDS2-0.22), 
           label=paste("Stress = ", round(sol_t8$stress,2)), hjust=0)+
  geom_path(data=df_ell_t8, aes(x=NMDS1, y=NMDS2, linetype=Treatment), size=1)+
  scale_linetype_manual(values=c("C"="solid","S4"="dotted"))+
  scale_colour_manual(values=c("C"="blue","S4"="orange"))+
  ggtitle("Trial 8, RWU")+
   geom_point(aes(shape=Treatment), size=4, alpha=0.8) + 
   scale_shape_manual(values=c(15,16)) + 
   theme_grey()+
   theme(plot.title=element_text(size = 16, face = "bold", hjust = 0.5))
NMDSplot_t8

```

## Plot all together


```{r betasummary, fig.width=10, height=12}

(NMDSplot_t1 | NMDSplot_t2 | NMDSplot_t3) /
(NMDSplot_t4 | NMDSplot_t7 | NMDSplot_t8) + 
   plot_layout(guides = 'collect') +
   plot_annotation(tag_levels = "A") &
   theme(legend.position='bottom')

ggsave("Figures/Figure3.png",  width=12, height=8, dpi = 400)

```


# Figure 4 - Upset plot

```{r}

library(VennDiagram)
library(UpSetR)
library(MicrobiotaProcess) # BiocManager::install("MicrobiotaProcess")
library(phylosmith) # devtools::install_github('schuyler-smith/phylosmith')

#Plot by Trial
upsetda <- get_upset(obj=justbacteria_12000_rarefy, factorNames="Trial")
By_Trial <- upset(upsetda, sets.bar.color = "skyblue", sets = c("T1", "T2", "T3", "T4", "T5", "T6","T7", "T8"), queries = list(list(query = intersects, params = list("T1", "T2", "T3", "T4", "T5","T6", 
     "T7", "T8"), color = "lightpink1", active = T)),order.by = "freq", empty.intersections = "on",keep.order = TRUE,number.angles = 0, point.size = 2.5,line.size = 1, mb.ratio = c(0.55, 0.45),
    mainbar.y.label = "ASVs Intersections", sets.x.label = "Total number of ASVs per trial",text.scale = c(1.3, 1.3, 1.3, 1.0, 1.2, 1.2))
By_Trial
ggsave("Figures/Figure4.png",  width=12, height=8, dpi = 400)

common_taxa(justbacteria_12000_rarefy, treatment = "Trial", n = 'all')
```

## ASV core abundance
```{r, fig.width=15, fig.height=10}

ASV_colors <- c("brown4","brown2","blue4","blue","chocolate2",
                   "aquamarine4","burlywood1","darkgoldenrod1",
                   "gold1","hotpink1","#808080","#94FFB5",
                   "#C84248","#8A7C64", "#599861","gray","yellow","blue","red4","green3","orange","gray0","cyan","chocolate4","pink4","grey85")
physeq.subset <- subset_taxa(justbacteria_12000_rarefy, rownames(tax_table(justbacteria_12000_rarefy)) %in% c("ASV_1","ASV_10",  "ASV_101", "ASV_103", "ASV_108", "ASV_11",  "ASV_120", "ASV_13",  "ASV_16", 
 "ASV_165", "ASV_18",  "ASV_199", "ASV_20",  "ASV_200", "ASV_22",  "ASV_276", "ASV_28",  "ASV_3",  
 "ASV_306", "ASV_33",  "ASV_35",  "ASV_37",  "ASV_38",  "ASV_40",  "ASV_415", "ASV_42",  "ASV_5",  
 "ASV_56",  "ASV_6",   "ASV_64",  "ASV_7",   "ASV_76",  "ASV_9",   "ASV_98"))

Coretaxa <- physeq.subset %>% psmelt() %>% 
   # calculate mean of the technical replicates per ASV
   group_by(Num, Treatment, Trial,Tank_Replicate, Hatchery, Name, OTU, Species) %>%
   summarise(mean_abun=mean(Abundance)) %>% ungroup() %>% 
   # select the 25 most abundant OTUs, then group all the Others
   mutate(ASVOther=forcats::fct_lump_n(f=OTU, w=mean_abun, other_level="Others", n=25)) 
   
Core_Order_plot <- Coretaxa %>% 
   # calculate sums for Other
   group_by(Num,Treatment,Name,Hatchery,ASVOther) %>% 
   summarise(sum_abun=sum(mean_abun)) %>% ungroup() %>% 
   # reorder OrderOther
   mutate(ASVOther = reorder(ASVOther, -sum_abun),
          ASVOther = fct_relevel(ASVOther, "Others", after = Inf)) %>% 
   # start plotting
   ggplot() +
   geom_col(aes(x = Num, y = sum_abun, fill = ASVOther), position="fill")+
   labs(y="Percent Core ASV Abundance", x=NULL, fill="Core ASV") +
   facet_nested(.~Hatchery+Name+Treatment, scales = "free", space="free", switch="x") +   
   scale_fill_manual(values = ASV_colors) +
   scale_y_continuous(expand = c(0,0),labels=scales::percent)+
   theme_minimal()+
   theme(axis.text.x = element_blank(), panel.spacing = unit(0.2,"lines"),
         panel.grid.major.x = element_blank(), panel.grid.minor.x = element_blank(),
         strip.background = element_rect(fill="gray85"),
         axis.text.y.left = element_text(size=16, face="bold"),
         axis.title.y = element_text(size = 18, face="bold"),
         strip.text.x = element_text(size = 18,face="bold"),
         legend.position="right", legend.title = element_text(size=20, face="bold"), 
         legend.text = element_text(size=18, face="bold"))

Core_Order_plot
# calculate percentage abundance of each ASV
Coretaxa <- physeq.subset %>% psmelt() %>% 
   group_by(OTU,Trial) %>%
   summarise(Abun=mean(Abundance/12640)*100) 
Abun_ASV <-spread(Coretaxa, OTU, Abun)
Abun_ASV2 <-Abun_ASV %>% remove_rownames %>% column_to_rownames(var="Trial")
Abun_ASV3 <-Abun_ASV2[,order(colSums(Abun_ASV2),decreasing=TRUE)]
Abun_ASV3

```

# Figure 5 - AncomBC enrichment

```{r, fig.width=6}

phy <- subset_samples(justbacteria_12000_rarefy, Trial %in% c("T1", "T2","T3","T4","T7","T8"))
out <- ANCOMBC::ancombc(phy,formula = "Trial+Treatment", p_adj_method = "fdr", lib_cut = 1000, group = "Treatment", 
                        struc_zero = TRUE, neg_lb = TRUE, tol = 1e-5, max_iter = 100, conserve = TRUE, alpha = 0.05, 
                        global = TRUE)

res <- out$res
Diff <- res$diff_abn
Diff1 <- Diff%>%
    data.frame() %>%
    select("TreatmentS4") %>%
    mutate_if(is.factor,as.character)
Diff2 <- cbind(FeatureID = rownames(Diff1 ), Diff1)
rownames(Diff2) <- NULL
names(Diff2)[names(Diff2) == "TreatmentS4"] <- "Differential" 

LFC <- res$beta
names(LFC)[names(LFC) == "TreatmentS4"] <- "Logfoldchange" 
LFC1 <- LFC%>%
    data.frame() %>%
    select("Logfoldchange") %>%
    mutate_if(is.factor,as.character)
LFC2 <- cbind(FeatureID = rownames(LFC1 ), LFC1)
rownames(LFC2) <- NULL

pval <- res$q_val
pval <- cbind(FeatureID = rownames(pval ), pval)
rownames(pval) <- NULL
names(pval)[names(pval) == "TreatmentS4"] <- "pvalue" 
Pvalue <-pval%>%
    data.frame() %>%
    select("FeatureID","pvalue") %>%
    mutate_if(is.factor,as.character)
stat <- full_join(LFC2, Pvalue)
stat1 <- full_join(stat,Diff2)

sigtab <- stat1 %>% filter(pvalue < 0.05)

##Combine ASV and Genus together into one column
ANCOM_BC1 <- sigtab %>% left_join(as.data.frame(phyloseq::tax_table(justbacteria_12000_rarefy)) %>% rownames_to_column("FeatureID")) %>% 
   mutate(Class=str_remove(Class, "c__"),
          Genus=str_remove(Genus, "g__"),
          Taxa=paste(FeatureID, Class, Genus, sep="_")) %>% 
   filter(abs(Logfoldchange)>0.3)

ANCOMBC_plot <- ggplot(ANCOM_BC1,aes(x=reorder(Taxa,-Logfoldchange), y=Logfoldchange,fill=Logfoldchange>0))+
   geom_col() + 
   coord_flip() +
   scale_fill_manual(values=c("blue","orange"),
                     labels=c("Control","S4")) +
   theme(legend.position = c(0.2, 0.15),
         text = element_text(size = 14, face="bold"),
         title = element_text(size = 16, face="bold", colour = "black"))+
   labs(fill="Enriched Group", y="Log fold change", x="ASV")
ANCOMBC_plot

ggsave(filename = "Figures/Figure5.png", width=10, height=6, dpi = 400)

```

# Figure 6 - Trial 8 UV vs NonUV Figures

```{r uvnonuv}

UV_NonUV <- subset_samples(justbacteria_12000_rarefy, UV_NUV == "Y")

UV_NonUV_SD <- sample_data(UV_NonUV) %>%
    data.frame() %>%
    select("Treatment","Trial", "Num","Type", "Name","Tank_Replicate","Condition") %>%
    mutate_if(is.factor,as.character)
UV_NonUV_SD1 <- cbind(SampleID = rownames(UV_NonUV_SD ), UV_NonUV_SD)
Meta <-UV_NonUV_SD1 %>%group_by(Num,Condition,Treatment,Type,Tank_Replicate)%>%summarise(avg_abundance = mean(sum)) %>% 
   remove_rownames %>% column_to_rownames(var="Num")
```

## Alpha-diversity

```{r uvnonuvdiv}

DIV <- data.frame(
  "Chao1" = phyloseq::estimate_richness(UV_NonUV, measures = "Chao1"),
  "Shannon" = phyloseq::estimate_richness(UV_NonUV, measures = "Shannon"),
  "Simpson" = phyloseq::estimate_richness(UV_NonUV, measures = "Simpson"),
  "Status" = phyloseq::sample_data(UV_NonUV)$Treatment)


DIV$Condition <- UV_NonUV_SD1$Condition
DIV$Num <- UV_NonUV_SD1$Num
DIV$Tank_Replicate <- UV_NonUV_SD1$Tank_Replicate
DIV$Treatment <- UV_NonUV_SD1$Treatment
DIV$Type <- UV_NonUV_SD1$Type

DIVS <- DIV %>% group_by(Num,Condition,Tank_Replicate, Type, Treatment) %>% summarise(mean_div=mean(Simpson), sd_div=sd(Simpson))
DIVS$Condition <- as.character(DIVS$Condition)
DIVS$Condition <- factor(DIVS$Condition, levels=unique(DIVS$Condition))
DIVS
write.csv(DIVS,"Tables/UV_DIV.csv")

Div_UV <- ggplot(DIVS, aes(as.factor(Condition), mean_div, fill=Treatment))+
   geom_boxplot(alpha=0.7)+
   geom_jitter(aes(shape=Treatment), width=0.2)+
   scale_fill_manual(values=c("blue", "orange"))+
   scale_shape_manual(values=c(22,21))+
  labs(y="Simpson's Diversity ", x="Treatment", fill="Treatment")+theme_bw()+
  theme(strip.text.x = element_text(size=12,face="bold"),
        strip.background = element_rect(fill="gray85"),
        panel.border = element_rect(fill=NA, colour = "black", size=1.5))+
  geom_signif(comparisons=list(c("C_NUV", "C_UV")), annotations="**", 
              y_position = 0.75, tip_length = -0.1, vjust=3)+
  geom_signif(comparisons=list(c("S4_NUV", "C_UV")), annotations="**", 
              y_position = 0.98, tip_length = 0.1, hjust=1)+
  theme(axis.text.y = element_text(size="12", color="black"), axis.title.y = element_text(face="bold",size="12", color="black"),
  axis.text.x = element_text(size="12", color="black", angle = 0), 
  axis.title.x.bottom = element_text(face="bold",size="14", color="black"),
  axis.text.x.top= element_text(face="bold",size="14", color="black"),
        legend.position = "none",panel.spacing = unit(0.2, "mm"),               
        panel.border = element_rect(colour="black", size=1))+
  scale_y_continuous(limits=c(0.6,1))
Div_UV

## Statistics
## Run Anova
Simpsons_Div <- aov(mean_div ~ Type*Treatment, data=DIVS)
summary(Simpsons_Div)
## Run Tukey's test for pairwise comparison
TukeyHSD(Simpsons_Div )

```

## NMDS beta-diversity

```{r}
 
ASV_NUV_UV <- subset_samples(justbacteria_12000_rarefy, UV_NUV == "Y") %>% psmelt() %>%
   group_by(Num,OTU) %>% summarise(avg_abundance = mean(Abundance)) %>%  # average by tank
   spread(Num, avg_abundance) %>% column_to_rownames(var="OTU") %>% t() %>% as.data.frame()
Meta_NUV_UV <- All_Meta[rownames(ASV_NUV_UV),] %>% 
   mutate(Type=word(Condition, 2, sep = "_"))

sol_NUV_UV <- metaMDS(ASV_NUV_UV, distance = "bray", k = 2)
NMDS_NUV_UV=data.frame(NMDS1=sol_NUV_UV$point[,1], NMDS2=sol_NUV_UV$point[,2], 
                    Condition=as.factor(Meta_NUV_UV$Condition),
                    Treatment=as.factor(Meta_NUV_UV$Treatment),
                    Type=as.factor(Meta_NUV_UV$Type))

plot.new()
ord_NUV<-ordiellipse(sol_NUV_UV, as.factor(Meta_NUV_UV$Condition), display = "sites", kind ="sd", conf = 0.95, label = T)
#dev.off()

df_ell_NUV <- data.frame()
for(g in levels(NMDS_NUV_UV$Condition)){
  if(g!="" && (g %in% names(ord_NUV))){
    df_ell_NUV <- rbind(df_ell_NUV, cbind(as.data.frame(with(NMDS_NUV_UV[NMDS_NUV_UV$Condition==g,],                            veganCovEllipse(ord_NUV[[g]]$cov,ord_NUV[[g]]$center,ord_NUV[[g]]$scale))),Condition=g))}}

#Calculate p-value:
adon_NUV_UV<-adonis2(ASV_NUV_UV ~ Type*Treatment, data=Meta_NUV_UV, method="bray")
adon_NUV_UV
# calculate beta-dispersion p-value:
disper_NUV_UV <- betadisper(vegdist(ASV_NUV_UV, distance="bray"), Meta_NUV_UV$Condition)
anova(disper_NUV_UV)


#Plot NMDS
NMDSplot_NUV_UV<-ggplot(data=NMDS_NUV_UV,aes(NMDS1,NMDS2))+
  annotate("text",x=min(NMDS_NUV_UV$NMDS1-0.5),y=min(NMDS_NUV_UV$NMDS2-0.3), 
           label=paste("p-value = ", round(adon_NUV_UV$`Pr(>F)`[1],3)), hjust=0)+
  annotate("text",x=min(NMDS_NUV_UV$NMDS1-0.5),y=min(NMDS_NUV_UV$NMDS2-0.22), 
           label=paste("Stress = ", round(sol_NUV_UV$stress,2)), hjust=0)+
  geom_path(data=df_ell_NUV, aes(x=NMDS1, y=NMDS2, linetype=Condition, color=Condition), size=1)+
  scale_linetype_manual(values=c("C_NUV"="solid","S4_NUV"="dotted", "C_UV"="solid","S4_UV"="dotted"))+
  scale_colour_manual(values=c("C_NUV"="blue","S4_NUV"="orange", "C_UV"="blue","S4_UV"="orange"))+
  geom_point(aes(shape=Condition, color=Condition), size=5) + scale_shape_manual(values=c(15,16, 15,16)) + 
   theme_grey()+
   theme(plot.title=element_text(size = 16, face = "bold", hjust = 0.5))
NMDSplot_NUV_UV

# install_github("pmartinezarbizu/pairwiseAdonis/pairwiseAdonis")

##Pairwise comparison between the conditions
pairwiseAdonis::pairwise.adonis2(ASV_NUV_UV ~Condition, data=Meta_NUV_UV, by=NULL,method="bray", k=2, nperm = 999)
##Pairwise comparison between the Water type
pairwiseAdonis::pairwise.adonis2(ASV_NUV_UV ~Type, data=Meta_NUV_UV, by=NULL,method="bray", k=2, nperm = 999)

```

# Figure S1 - Rarefaction curves

## All data

```{r rarefactioncurve1}
# number of ASVs per sample
(S <- specnumber(t(otu_table(justbacteria))))
# smallest number of reads in a sample
(raremax <- min(rowSums(t(otu_table(justbacteria)))))
# rarefied number of taxa per sample
Srare <- vegan::rarefy(t(otu_table(justbacteria)), raremax)
# slope at the end of the rarefacetion curve per sample
Sslope <- vegan::rareslope(t(otu_table(justbacteria)), raremax)

# plot observed vs rarefied number of ASVs
plot(S, Srare, xlab = "Observed No. of ASVs", ylab = "Rarefied No. of ASVs")
abline(0,1)
# plot slopes
plot(S, Sslope, xlab = "Observed No. of ASVs", ylab = "Slope at rarefied sample size")

# store rarefaction curves data with 100 steps
rarecurve_data <- rarecurve(t(otu_table(justbacteria)), step = 100, sample = raremax)

# clean plot of rarefaction curves
rarecurvealldata <- map_dfr(rarecurve_data, bind_rows) %>%
   bind_cols(SampleID = rownames(t(otu_table(justbacteria))),.) %>%
   pivot_longer(-SampleID) %>%
   drop_na() %>%
   mutate(n_seqs = as.numeric(str_replace(name, "N", ""))) %>%
   select(-name) %>%
   left_join((meta_table %>% rownames_to_column("SampleID"))) %>%
   ggplot(aes(x=n_seqs, y=value, group=SampleID, color=Trial)) +
   geom_line(lwd=0.5) +
   geom_vline(xintercept = 12640, lty="dotted")+
   scale_color_manual(values = c("#c95795", "#70a845", "#8475ca", "#b79040","#4aad92", "#ca5b49", "navy", "tan4"))+
   scale_x_continuous(labels=scales::label_comma())+
   scale_y_continuous(limits=c(0,400))+
   theme(legend.key.width = unit(1.33,"cm"), legend.position="right")+
   labs(x = "Number of sequences", y="Number of ASVs detected", color=NULL, lty=NULL, title="All data")
rarecurvealldata

```


## Rarefied data

```{r rarefactioncurve2}
# number of ASVs per sample
(S <- specnumber(t(otu_table(justbacteria_12000_rarefy))))
# smallest number of reads in a sample
(raremax <- min(rowSums(t(otu_table(justbacteria_12000_rarefy)))))
# rarefied number of taxa per sample
Srare <- vegan::rarefy(t(otu_table(justbacteria_12000_rarefy)), raremax)
# slope at the end of the rarefacetion curve per sample
Sslope <- vegan::rareslope(t(otu_table(justbacteria_12000_rarefy)), raremax)

# plot observed vs rarefied number of ASVs
plot(S, Srare, xlab = "Observed No. of ASVs", ylab = "Rarefied No. of ASVs")
abline(0,1)
# plot slopes
plot(S, Sslope, xlab = "Observed No. of ASVs", ylab = "Slope at rarefied sample size")

# store rarefaction curves data with 100 steps
rarecurve_data <- rarecurve(t(otu_table(justbacteria_12000_rarefy)), step = 100, sample = raremax)

# clean plot of rarefaction curves
rarecurveraredata <- map_dfr(rarecurve_data, bind_rows) %>% 
   bind_cols(SampleID = rownames(t(otu_table(justbacteria_12000_rarefy))),.) %>%
   pivot_longer(-SampleID) %>%
   drop_na() %>%
   mutate(n_seqs = as.numeric(str_replace(name, "N", ""))) %>%
   select(-name) %>%
   left_join((meta_table %>% rownames_to_column("SampleID"))) %>% 
   ggplot(aes(x=n_seqs, y=value, group=SampleID, color=Trial)) +
   geom_line(lwd=0.5) +
   geom_vline(xintercept = 12640, lty="dotted")+
   scale_color_manual(values = c("#c95795", "#70a845", "#8475ca", "#b79040","#4aad92", "#ca5b49", "navy", "tan4"))+
   scale_x_continuous(labels=scales::label_comma())+
   scale_y_continuous(limits=c(0,400))+
   theme(legend.key.width = unit(1.33,"cm"), legend.position="right")+
   labs(x = "Number of sequences", y="Number of ASVs detected", color=NULL, lty=NULL, title="Rarefied data")
rarecurveraredata

```

## Plot together

```{r rarecurvesummary}

rarecurvealldata / rarecurveraredata + plot_annotation(tag_levels="A") +
   plot_layout(guides="collect") &
   theme(legend.justification =  c(0.5,1))

ggsave("Figures/FigureS1.png", width=6, height=8, dpi=400)

```

# Figure S2 - Phyla barplot

```{r phylabarplot, fig.width=15}

Phylum_colors <- c("#4DAA57","#F1A66A","#F26157", "#F9ECCC", "#679289", "#33658A","#F6AE2D","#86BBD8")

Phylum_abundance <- justbacteria_12000_rarefy %>% tax_glom(taxrank = "Phylum") %>% 
   psmelt() %>% 
   # calculate mean of the technical replicates per Phylum
   group_by(Num,Treatment,Name,Hatchery,Phylum) %>% 
   summarise(mean_abun=mean(Abundance/12640*100)) %>% ungroup() %>% 
   # make new column of cleaned top 12 Phylum
   mutate(PhylumClean=str_remove(Phylum, "p__"),
          # select the 5 most abundant phyla, then group all the Others
          PhylumOther=forcats::fct_lump_n(f=PhylumClean, w=mean_abun, other_level="Others", n=5)) %>% 
   # calculate sums for Other
   group_by(Num,Treatment,Name,Hatchery,PhylumOther) %>% 
   summarise(sum_abun=sum(mean_abun)) %>% ungroup() %>% 
   # reorder OrderOther
   mutate(PhylumOther = reorder(PhylumOther, -sum_abun),
          PhylumOther = fct_relevel(PhylumOther, "Others", after = Inf)) 
   # start plotting
   Phylum_abundance %>% ggplot() +
   geom_col(aes(x = Num, y = sum_abun, fill = PhylumOther), position="fill")+
   labs(y="Percent Phylum Abundance", x=NULL, fill="Order") +
   facet_nested(.~Hatchery+Name+Treatment, scales = "free", space="free", switch="x") +
   scale_fill_manual(values = Phylum_colors) +
   scale_y_continuous(expand = c(0,0),labels=scales::percent)+
   theme_minimal()+
   theme(axis.text.x = element_blank(), panel.spacing = unit(0.2,"lines"),
         panel.grid.major.x = element_blank(), panel.grid.minor.x = element_blank(),
         strip.background = element_rect(fill="gray85"),
         axis.text.y.left = element_text(size=16, face="bold"),
         axis.title.y = element_text(size = 18, face="bold"),
         strip.text.x = element_text(size = 18,face="bold"),
         legend.position="bottom", legend.title = element_text(size=20, face="bold"), 
         legend.text = element_text(size=18, face="bold"))

ggsave("Figures/FigureS2.png", width=18, height=8, dpi=400)

Phylum_abundance 
write.csv(Phylum_abundance,"Tables/Phylum.csv")
 
```


# Figure S3 - Chao richness
```{r chao}
##Subset samples to remove Trial 5 and trial6
phy <- subset_samples(justbacteria_12000_rarefy, Trial %in% c("T1", "T2","T3","T4","T7","T8"))

SD <- sample_data(phy) %>%
    data.frame() %>%
    select("Treatment","Trial","Num","Tank_Replicate") %>% 
  mutate_if(is.factor,as.character)
SD1 <- cbind(SampleID = rownames(SD ), SD)
rownames(SD1 ) <- NULL
adiv <- data.frame(
  "Observed" = phyloseq::estimate_richness(phy, measures = "Observed"),
  "Simpson" = phyloseq::estimate_richness(phy, measures = "Simpson"),
  "Chao1"= phyloseq::estimate_richness(phy, measures = "Chao1"),
  "Status" = phyloseq::sample_data(phy)$Treatment)
adiv$Trial <- SD$Trial
adiv$Treatment <- SD$Treatment
adiv$Num <- SD$Num
adiv$Tank_Replicate<- SD$Tank_Replicate

adivC <- adiv %>% group_by(Num,Trial,Treatment,Tank_Replicate) %>% summarise(mean_Chao1=mean(Chao1.Chao1), sd_read=sd(Chao1.Chao1))
DivC <-adivC %>% remove_rownames %>% column_to_rownames(var="Num")

Div_Chao1 <- ggplot(DivC, aes(x=Treatment, y=mean_Chao1, fill=Treatment))+
   geom_boxplot(alpha=0.7)+
   geom_jitter(aes(shape=Treatment), width=0.2)+
   scale_fill_manual(values=c("blue", "orange"))+
   scale_shape_manual(values=c(22,21))+
  facet_grid(.~Trial, scales="free",space="free_x",switch="y")+
  theme_gray()+
  labs(y="Chao1 Richness", x="Treatment", fill="Treatment")+theme_bw()+
   scale_y_continuous(limits=c(0,NA))+
  theme(strip.text.x = element_text(size=12,face="bold"),
        panel.grid.major.x = element_blank(),
        strip.background = element_rect(fill="gray85"),
        axis.text.y = element_text(size="14", color="black"), 
        axis.title.y = element_text(face="bold",size="14", color="black"),
        axis.text.x = element_text(size="12", color="black", angle = 0), 
        axis.title.x.bottom = element_text(face="bold",size="14", color="black"),
        axis.text.x.top= element_text(face="bold",size="14", color="black"),
        legend.position = "None",panel.spacing = unit(0.2, "mm"),               
        panel.border = element_rect(colour="grey"))
Div_Chao1

ggsave(filename = "Figures/FigureS3.png", width=7, height=4, dpi = 400)

##Stats
##Chao1 richness
Chao1_Anova <- aov(mean_Chao1 ~ Treatment*Trial, data=DivC)
summary(Chao1_Anova)
tukey_Chao1 <- HSD.test(Chao1_Anova, "Trial", group = FALSE)
tukey_Chao1$comparison
```


# Figure S4 - Beta diversity by Trial


```{r betadivtrial,fig.width=4}
ASV_all <- subset_samples(justbacteria_12000_rarefy, Trial %in% c("T1", "T2","T3","T4","T7","T8")) %>% psmelt() %>%
   group_by(Num,OTU) %>% summarise(avg_abundance = mean(Abundance)) %>%  # average by tank
   spread(Num, avg_abundance) %>% column_to_rownames(var="OTU") %>% t() %>% as.data.frame() %>% 
   rownames_to_column("Num") %>% mutate(Num=as.numeric(Num)) %>% arrange(Num)
Meta_all <- All_Meta %>% filter(Trial %in% c("T1", "T2","T3","T4","T7","T8")) %>% 
   rownames_to_column("Num") %>% mutate(Num=as.numeric(Num)) %>% arrange(Num)

sol_all <- metaMDS(ASV_all, distance = "bray", k = 2)
NMDS_all=data.frame(NMDS1=sol_all$point[,1], NMDS2=sol_all$point[,2], 
                   Treatment=as.factor(Meta_all$Treatment),
                   Trial=as.factor(Meta_all$Trial))

plot.new()
ord_trial<-ordiellipse(sol_all, as.factor(Meta_all$Trial), display = "sites", kind ="sd", conf = 0.95, label = T)
#dev.off()

df_ell_trial <- data.frame()
for(g in levels(NMDS_all$Trial)){
  if(g!="" && (g %in% names(ord_trial))){
    df_ell_trial <- rbind(df_ell_trial, cbind(as.data.frame(with(NMDS_all[NMDS_all$Trial==g,],                            veganCovEllipse(ord_trial[[g]]$cov,ord_trial[[g]]$center,ord_trial[[g]]$scale))),Trial=g))}}

#Calculate p-value:
adon_trial<-adonis2(ASV_all ~Trial, data=Meta_all, by=NULL,method="bray", k=2)
adon_trial
# calculate beta-dispersion p-value:
disper_trial <- betadisper(vegdist(ASV_all, distance="bray"), Meta_all$Trial)
anova(disper_trial)
plot(disper_trial)

#Plot NMDS
NMDSplot_trial<-ggplot(data=NMDS_all,aes(NMDS1,NMDS2,col=Trial))+
  annotate("text",x=-4,y=3.5, 
           label=paste("p-value = ", round(adon_trial$`Pr(>F)`[1],3)), hjust=0)+
  annotate("text",x=-4,y=3.1, 
          label=paste("Stress = ", round(sol_all$stress,2)), hjust=0)+
  geom_path(data=df_ell_trial, aes(x=NMDS1, y=NMDS2), size=1)+
  scale_colour_manual(values=c("#c95795", "#70a845", "#8475ca", "#b79040","#4aad92", "#ca5b49"))+
  ggtitle("Trial")+
  geom_point(aes(shape=Treatment), size=3, alpha=0.8) + scale_shape_manual(values=c(15,16)) + 
   theme_grey()+
   theme(plot.title=element_text(size = 16, face = "bold", hjust = 0.5))

ggsave("Figures/FigureS4.png",width=5, height=4, dpi = 400)

NMDSplot_trial
```


## Statistics

```{r}

#PERMANOVA
Meta2 <- All_Meta %>% filter(Trial != "T5" & Trial != "T6") %>%  ## remove trial5 and trial6
   unite("TrialTreatment", Trial, Treatment, remove = FALSE)

phy <- subset_samples(justbacteria_12000_rarefy, Trial %in% c("T1", "T2","T3","T4","T7","T8"))
all_order <- phy %>% psmelt() %>%group_by(Num,OTU)%>%summarise(avg_abundance = mean(Abundance))
Order <- spread(all_order, Num,avg_abundance) %>% remove_rownames %>% column_to_rownames(var="OTU")
Abun <- as.data.frame(t(Order))

# adonis2 PERMANOVA
BC.dist=vegdist(Abun, distance="bray")
Permanova <-adonis2(BC.dist ~ Treatment*Trial, data = Meta2, permutations = 1000)
Permanova

# calculate beta-dispersion p-value:
disperTreat <- betadisper(BC.dist, c(Meta2$Treatment))
anova(disperTreat)
disperTrial <- betadisper(BC.dist, c(Meta2$Trial))
anova(disperTrial)

```

# Figure S5 - Shared ASVs
```{r}
##Plot by treatment
upsetda <- get_upset(obj=justbacteria_12000_rarefy, factorNames="Treatment")
By_Treatment <- upset(upsetda, sets=c("S4", "C"), sets.bar.color = c("orange","blue"),order.by = "freq", empty.intersections = "on",keep.order = TRUE,number.angles = 0, point.size = 2.5,line.size = 1,mainbar.y.label = "ASVs Intersections", sets.x.label = "ASV Per Treatment",text.scale = c(2, 2, 1.3, 1.3, 2, 2.5))

By_Treatment
ggsave("Figures/FigureS5.png",width=0.5, height=1, dpi = 400)
```


# Figure S6 - Vibrio diversity
## Diversity 
```{r}
##Simpson's diversity
phy <- subset_samples(justbacteria_12000_rarefy, Trial %in% c("T1", "T2","T3","T4","T7","T8"))
vibrio1 <- subset_taxa(phy, Order== "o__Vibrionales")
div_Vib <- data.frame(
  "Simpson" = phyloseq::estimate_richness(vibrio1, measures = "Simpson"),
  "Chao1"= phyloseq::estimate_richness(vibrio1, measures = "Chao1"),
  "Status" = phyloseq::sample_data(vibrio1)$Treatment)

div_Vib$Trial <-as.character(SD$Trial)
div_Vib$Treatment <-as.character(SD$Treatment)
div_Vib$Num <-as.character(SD$Num)
div_Vib$Tank_Replicate <-as.character(SD$Tank_Replicate)

div_Vib1 <- div_Vib %>% group_by(Num,Treatment, Trial, Tank_Replicate) %>% summarise(mean_div=mean(Simpson), sd_div=sd(Simpson))
div_Vib1
write.csv(div_Vib1,"Tables/Vibrio_diversity.csv")

Div_Vibrio <- ggplot(div_Vib1, aes(as.factor(Treatment),mean_div, fill=Treatment)) +
   geom_boxplot(alpha=0.7)+
   ggbeeswarm::geom_beeswarm(aes(shape=Treatment))+
   scale_fill_manual(values=c("blue", "orange"))+
   scale_shape_manual(values=c(22,21))+
  labs(y="Vibrio Simpson's Diversity", x="Treatment", fill="Treatment")+theme_bw()+
  theme(strip.text.x = element_text(size=12,face="bold"),
        panel.grid.major.x = element_blank(),
        strip.background = element_rect(fill="gray85"),
        axis.text.y = element_text(size="14", color="black"), 
        axis.title.y = element_text(face="bold",size="14", color="black"),
        axis.text.x = element_text(size="12", color="black", angle = 0), 
        axis.title.x.bottom = element_text(face="bold",size="14", color="black"),
        axis.text.x.top= element_text(face="bold",size="14", color="black"),
        legend.position = "None",panel.spacing = unit(0.2, "mm"))+
  scale_y_continuous(limits=c(0,1))
Div_Vibrio

## Run the model
Simpsons_Div <- aov(mean_div ~ Trial*Treatment, data=div_Vib1)
summary(Simpsons_Div)

```

## Abundance

```{r abundance}

## Vibrio Abundance

vibrioabundances <- vibrio1 %>% tax_glom(taxrank = "Order") %>% psmelt() %>% 
   group_by(Num, Treatment, Trial,Tank_Replicate, Hatchery, Name) %>%
   summarise(VibrioAbundance=mean(Abundance)) %>% 
   # convert to percent just for plotting based on rarefied counts
   mutate(percent=(VibrioAbundance/12640)*100)

vibrioabundances

Vibrio_abun <- vibrioabundances %>% 
   ggplot(aes(x = Treatment, y = percent, by=Treatment, fill= Treatment)) + 
   geom_boxplot(alpha=0.7)+
   ggbeeswarm::geom_beeswarm(aes(shape=Treatment))+
   scale_fill_manual(values=c("blue", "orange"))+
   scale_shape_manual(values=c(22,21))+
   labs(y=" Percent Vibrio Abundance", x="Treatment") +
      scale_y_continuous(labels=scales::label_percent(scale=1), limits=c(0,NA))+
   theme_bw()+
  theme(strip.text.x = element_text(size=12,face="bold"),
        panel.grid.major.x = element_blank(),
        strip.background = element_rect(fill="gray85"),
        axis.text.y = element_text(size="14", color="black"), 
        axis.title.y = element_text(face="bold",size="14", color="black"),
        axis.text.x = element_text(size="12", color="black", angle = 0), 
        axis.title.x.bottom = element_text(face="bold",size="14", color="black"),
        axis.text.x.top= element_text(face="bold",size="14", color="black"),
        legend.position = "None",panel.spacing = unit(0.2, "mm"))
Vibrio_abun

##Statistics
## Run the model
Vib_Abun <- aov(VibrioAbundance ~ Trial*Treatment, data=vibrioabundances)
summary(Vib_Abun)
## Run Tukey's test
tukey_Vib <- HSD.test(Vib_Abun, "Trial", group = FALSE)
tukey_Vib$comparison
```

## NMDS

### Trial 1

```{r nmdsvibrio1}

ASV_t1_vibrio <- subset_samples(justbacteria_12000_rarefy, Trial=="T1") %>% subset_taxa(Order== "o__Vibrionales") %>% psmelt() %>%
   group_by(Num,OTU) %>% summarise(avg_abundance = mean(Abundance)) %>%  # average by tank
   spread(Num,avg_abundance)  %>% remove_rownames %>% column_to_rownames(var="OTU") %>%  t() %>% as.data.frame()
Meta_t1_vibrio <- All_Meta %>% filter(Trial=="T1")

sol_t1_vibrio <-metaMDS(ASV_t1_vibrio ,distance = "bray", k = 2, trymax = 50)
NMDS_t1_vibrio=data.frame(NMDS1=sol_t1_vibrio$point[,1], NMDS2=sol_t1_vibrio$point[,2], Treatment=as.factor(Meta_t1_vibrio[,1]))

plot.new()
ord_t1_vibrio<-ordiellipse(sol_t1_vibrio, as.factor(Meta_t1_vibrio$Treatment), display = "sites", kind ="sd", conf = 0.95, label = T)
#dev.off()

df_ell_t1_vibrio <- data.frame()
for(g in levels(NMDS_t1_vibrio$Treatment)){
  if(g!="" && (g %in% names(ord_t1_vibrio))){
    df_ell_t1_vibrio <- rbind(df_ell_t1_vibrio, cbind(as.data.frame(with(NMDS_t1_vibrio[NMDS_t1_vibrio$Treatment==g,],                            veganCovEllipse(ord_t1_vibrio[[g]]$cov,ord_t1_vibrio[[g]]$center,ord_t1_vibrio[[g]]$scale))),Treatment=g))}}

#Calculate p-value:
adon_t1_vibrio<-adonis2(ASV_t1_vibrio ~Treatment, data=Meta_t1_vibrio, by=NULL,method="bray", k=2)
adon_t1_vibrio
# calculate beta-dispersion p-value:
disper_t1_vibrio <- betadisper(vegdist(ASV_t1_vibrio, distance="bray"), Meta_t1_vibrio$Treatment)
anova(disper_t1_vibrio)

#Plot NMDS
NMDSplot_t1_vibrio<-ggplot(data=NMDS_t1_vibrio,aes(NMDS1,NMDS2,col=Treatment))+
  annotate("text",x=min(NMDS_t1_vibrio$NMDS1-0.2),y=min(NMDS_t1_vibrio$NMDS2-0.6), 
           label=paste("p-value = ", round(adon_t1_vibrio$`Pr(>F)`[1],2)), hjust=0)+
  annotate("text",x=min(NMDS_t1_vibrio$NMDS1-0.2),y=min(NMDS_t1_vibrio$NMDS2-0.45), 
          label=paste("Stress = ", round(sol_t1_vibrio$stress,2)), hjust=0)+
  geom_path(data=df_ell_t1_vibrio, aes(x=NMDS1, y=NMDS2, linetype=Treatment), size=1)+
  scale_linetype_manual(values=c("C"="solid","S4"="dotted"))+
  scale_colour_manual(values=c("C"="blue","S4"="orange"))+
  ggtitle("Trial 1 Vibrio")+
  geom_point(aes(shape=Treatment), size=5) + scale_shape_manual(values=c(15,16)) + 
   theme_grey()+
   theme(plot.title=element_text(size = 16, face = "bold", hjust = 0.5))

NMDSplot_t1_vibrio

```


### Trial 4

```{r nmdsvibrio2}

ASV_t4_vibrio <- subset_samples(justbacteria_12000_rarefy, Trial=="T4") %>% subset_taxa(Order== "o__Vibrionales") %>% psmelt() %>%
   group_by(Num,OTU) %>% summarise(avg_abundance = mean(Abundance)) %>%  # average by tank
   spread(Num,avg_abundance)  %>% remove_rownames %>% column_to_rownames(var="OTU") %>%  t() %>% as.data.frame()
Meta_t4_vibrio <- All_Meta %>% filter(Trial=="T4")

sol_t4_vibrio <-metaMDS(ASV_t4_vibrio ,distance = "bray", k = 2, trymax = 50)
NMDS_t4_vibrio=data.frame(NMDS1=sol_t4_vibrio$point[,1], NMDS2=sol_t4_vibrio$point[,2], Treatment=as.factor(Meta_t4_vibrio[,1]))

plot.new()
ord_t4_vibrio<-ordiellipse(sol_t4_vibrio, as.factor(Meta_t4_vibrio$Treatment), display = "sites", kind ="sd", conf = 0.95, label = T)
#dev.off()

df_ell_t4_vibrio <- data.frame()
for(g in levels(NMDS_t4_vibrio$Treatment)){
  if(g!="" && (g %in% names(ord_t4_vibrio))){
    df_ell_t4_vibrio <- rbind(df_ell_t4_vibrio, cbind(as.data.frame(with(NMDS_t4_vibrio[NMDS_t4_vibrio$Treatment==g,],                            veganCovEllipse(ord_t4_vibrio[[g]]$cov,ord_t4_vibrio[[g]]$center,ord_t4_vibrio[[g]]$scale))),Treatment=g))}}

#Calculate p-value:
adon_t4_vibrio<-adonis2(ASV_t4_vibrio ~Treatment, data=Meta_t4_vibrio, by=NULL,method="bray", k=2)
adon_t4_vibrio
# calculate beta-dispersion p-value:
disper_t4_vibrio <- betadisper(vegdist(ASV_t4_vibrio, distance="bray"), Meta_t4_vibrio$Treatment)
anova(disper_t4_vibrio)


#Plot NMDS
NMDSplot_t4_vibrio<-ggplot(data=NMDS_t4_vibrio,aes(NMDS1,NMDS2,col=Treatment))+
  annotate("text",x=min(NMDS_t4_vibrio$NMDS1-0.4),y=min(NMDS_t4_vibrio$NMDS2-0.55), 
           label=paste("p-value = ", round(adon_t4_vibrio$`Pr(>F)`[1],2)), hjust=0)+
  annotate("text",x=min(NMDS_t4_vibrio$NMDS1-0.4),y=min(NMDS_t4_vibrio$NMDS2-0.45), 
          label=paste("Stress = ", round(sol_t4_vibrio$stress,2)), hjust=0)+
  geom_path(data=df_ell_t4_vibrio, aes(x=NMDS1, y=NMDS2, linetype=Treatment), size=1)+
  scale_linetype_manual(values=c("C"="solid","S4"="dotted"))+
  scale_colour_manual(values=c("C"="blue","S4"="orange"))+
  ggtitle("Trial 4 Vibrio")+
  geom_point(aes(shape=Treatment), size=5) + scale_shape_manual(values=c(15,16)) + 
   theme_grey()+
   theme(plot.title=element_text(size = 16, face = "bold", hjust = 0.5))

NMDSplot_t4_vibrio

```

## Plot together

```{r, fig.width=10, fig.height=8}

(Div_Vibrio + Vibrio_abun) / (NMDSplot_t1_vibrio + NMDSplot_t4_vibrio) + plot_layout(guides="collect") + plot_annotation(tag_levels = "A")

ggsave("Figures/FigureS6.png", width=8, height=7)

```


# Figure S7 - Vibrio abundance and composition


```{r, fig.width=15, fig.height=10}
vibrio_colors <- c("brown4","brown2","blue4","blue","chocolate2",
                   "aquamarine4","burlywood1","darkgoldenrod1",
                   "gold1","hotpink1","#808080","#94FFB5",
                   "#C84248","#8A7C64", "#599861","gray")

vibrioall <- subset_taxa(justbacteria_12000_rarefy, Order== "o__Vibrionales") 

vibriotaxa <- vibrioall %>% psmelt() %>% 
   # calculate mean of the technical replicates per ASV
   group_by(Num, Treatment, Trial,Tank_Replicate, Hatchery, Name, OTU, Species) %>%
   summarise(mean_abun=mean(Abundance)) %>% ungroup() %>% 
   # select the 15 most abundant OTUs, then group all the Others
   mutate(ASVOther=forcats::fct_lump_n(f=OTU, w=mean_abun, other_level="Others", n=13)) 
   
vibrio_Order_plot <- vibriotaxa %>% 
   # calculate sums for Other
   group_by(Num,Treatment,Name,Hatchery,ASVOther) %>% 
   summarise(sum_abun=sum(mean_abun)) %>% ungroup() %>% 
   # reorder OrderOther
   mutate(ASVOther = reorder(ASVOther, -sum_abun),
          ASVOther = fct_relevel(ASVOther, "Others", after = Inf)) %>% 
   # start plotting
   ggplot() +
   geom_col(aes(x = Num, y = sum_abun, fill = ASVOther), position="fill")+
   labs(y="Percent Vibrio ASV Abundance", x=NULL, fill="Vibrio ASV") +
   facet_nested(.~Hatchery+Name+Treatment, scales = "free", space="free", switch="x") +   
   scale_fill_manual(values = vibrio_colors) +
   scale_y_continuous(expand = c(0,0),labels=scales::percent)+
   theme_minimal()+
   theme(axis.text.x = element_blank(), panel.spacing = unit(0.2,"lines"),
         panel.grid.major.x = element_blank(), panel.grid.minor.x = element_blank(),
         strip.background = element_rect(fill="gray85"),
         axis.text.y.left = element_text(size=16, face="bold"),
         axis.title.y = element_text(size = 18, face="bold"),
         strip.text.x = element_text(size = 18,face="bold"),
         legend.position="right", legend.title = element_text(size=20, face="bold"), 
         legend.text = element_text(size=18, face="bold"))

vibrioabundances <- vibrioall %>% tax_glom(taxrank = "Order") %>% 
   psmelt() %>% 
   group_by(Num, Treatment, Trial,Tank_Replicate, Hatchery, Name) %>% 
   summarise(VibrioAbundance=mean(Abundance),
             VibrioPercent=VibrioAbundance/12640*100)
vibrioabundances
#write.xlsx(vibrioabundances, "VibrioAbundances.xlsx")

vibrio_Read_plot <- vibrioabundances %>% 
   ggplot(aes(Num,VibrioAbundance, fill=Treatment))+
   geom_col()+theme_grey()+
   facet_nested(.~Hatchery+Name+Treatment, scales = "free", space="free", switch="x") +
   labs(y="Normalized \nVibrio Abundance", x=NULL)+
   scale_fill_manual(values=c("blue","orange"))+
   scale_y_continuous(expand = c(0,0),labels=scales::label_comma())+
   theme_minimal()+
   theme(axis.text.x = element_blank(), panel.spacing = unit(0.2,"lines"),
         strip.background = element_rect(fill="gray85"),
         panel.grid.minor.x = element_blank(), panel.grid.major.x = element_blank(),
         axis.text.y.left = element_text(size=16, face="bold"),
         axis.title.y = element_text(size = 18, face="bold"),
         strip.text = element_blank(),
         legend.position="right", legend.title = element_text(size=20, face="bold"), 
         legend.text = element_text(size=18, face="bold"))

vibrio_Read_plot / vibrio_Order_plot + plot_layout(guides="collect", heights=c(1,2.5)) + plot_annotation(tag_levels = "A")

ggsave("Figures/FigureS7.png", width=19, height=9, dpi=400)
```


# Figure S8 - UV vs nonUV ancombc

## UV

```{r}
UV <- subset_samples(UV_NonUV, Type=="UV")
Orderabundance <- UV %>%
   tax_glom(taxrank = "Order")

out <- ANCOMBC::ancombc(Orderabundance,formula = "Treatment", p_adj_method = "fdr", lib_cut = 1000, group = "Treatment", 
                        struc_zero = TRUE, neg_lb = TRUE, tol = 1e-5, max_iter = 100, conserve = TRUE, alpha = 0.05, 
                        global = TRUE)
res <- out$res
Diff <- res$diff_abn
Diff1 <- Diff%>%
   data.frame() %>%
   select("TreatmentS4") %>%
   mutate_if(is.factor,as.character)
Diff2 <- cbind(FeatureID = rownames(Diff1 ), Diff1)
rownames(Diff2) <- NULL
names(Diff2)[names(Diff2) == "TreatmentS4"] <- "Differential" 

LFC <- res$beta
names(LFC)[names(LFC) == "TreatmentS4"] <- "Logfoldchange" 
LFC1 <- LFC%>%
   data.frame() %>%
   select("Logfoldchange") %>%
   mutate_if(is.factor,as.character)
LFC2 <- cbind(FeatureID = rownames(LFC1 ), LFC1)
rownames(LFC2) <- NULL
pval <- res$q_val
pval <- cbind(FeatureID = rownames(pval ), pval)
rownames(pval) <- NULL
names(pval)[names(pval) == "TreatmentS4"] <- "pvalue" 
Pvalue <-pval%>%
   data.frame() %>%
   select("FeatureID","pvalue") %>%
   mutate_if(is.factor,as.character)


sigtab <- full_join(LFC2, Pvalue) %>% 
   full_join(Diff2) %>% 
   filter(pvalue < 0.05)

##Reorder the logfoldchange in order
UV_df <- full_join(sigtab, as.data.frame(phyloseq::tax_table(justbacteria_12000_rarefy)) %>% rownames_to_column("FeatureID")) %>% 
   arrange(desc(Logfoldchange)) %>% 
   mutate(Order=str_remove(Order, "o__"))%>% 
   filter(abs(Logfoldchange)>0.3)

UV_df$Order <- factor(UV_df$Order, levels=unique(UV_df$Order))
UV_df

UV_df_plotA <- ggplot(UV_df,aes(x=Order,y=Logfoldchange,fill=Logfoldchange>0))+
   geom_col() + coord_flip()+
   scale_fill_manual(values=c("blue","orange"),
                     labels=c("Control","S4"))+
   ggtitle("UV") + theme(axis.text.x = element_text(colour="black", size=12, face="bold")) +
   theme(strip.background = element_rect(fill="gray85"),
         axis.text.y.left = element_text(colour="black", size=12, face="bold"),
         axis.text.y.right = element_text(colour="black", size=12, face="bold"),
         axis.title.y = element_text(size = 12, face="bold"),
         axis.title.x = element_text(size = 12, face="bold", colour = "black"),title = element_text(size = 12, face="bold"))+
   guides(fill=guide_legend(title="Enriched Group"))
UV_df_plotA

```

## Non-UV

```{r, fig.width=10}
NUV <- subset_samples(UV_NonUV, Type=="NUV")
Orderabundance <- NUV %>%
tax_glom(taxrank = "Order")
out <- ANCOMBC::ancombc(Orderabundance,formula = "Treatment", p_adj_method = "fdr", lib_cut = 1000, group = "Treatment", struc_zero = TRUE, neg_lb = TRUE, tol = 1e-5, max_iter = 100, conserve = TRUE, alpha = 0.05, 
  global = TRUE 
)
res <- out$res
Diff <- res$diff_abn
Diff1 <- Diff%>%
    data.frame() %>%
    select("TreatmentS4") %>%
    mutate_if(is.factor,as.character)
Diff2 <- cbind(FeatureID = rownames(Diff1 ), Diff1)
rownames(Diff2) <- NULL
names(Diff2)[names(Diff2) == "TreatmentS4"] <- "Differential" 

LFC <- res$beta
names(LFC)[names(LFC) == "TreatmentS4"] <- "Logfoldchange" 
LFC1 <- LFC%>%
    data.frame() %>%
    select("Logfoldchange") %>%
    mutate_if(is.factor,as.character)
LFC2 <- cbind(FeatureID = rownames(LFC1 ), LFC1)
rownames(LFC2) <- NULL
pval <- res$q_val
pval <- cbind(FeatureID = rownames(pval ), pval)
rownames(pval) <- NULL
names(pval)[names(pval) == "TreatmentS4"] <- "pvalue" 
Pvalue <-pval%>%
    data.frame() %>%
    select("FeatureID","pvalue") %>%
    mutate_if(is.factor,as.character)

sigtab <- full_join(LFC2, Pvalue) %>% 
   full_join(Diff2) %>% 
   filter(pvalue < 0.05)

##Reorder the logfoldchange in order
NUV_df <- full_join(sigtab, as.data.frame(phyloseq::tax_table(justbacteria_12000_rarefy)) %>% rownames_to_column("FeatureID")) %>% 
   arrange(desc(Logfoldchange)) %>% 
   mutate(Order=str_remove(Order, "o__"))%>% 
   filter(abs(Logfoldchange)>0.3)

NUV_df$Order <- factor(NUV_df$Order, levels=unique(NUV_df$Order))
NUV_df

NUV_df_plotB <- ggplot(NUV_df,aes(x=Order,y=Logfoldchange,fill=Logfoldchange>0))+
   geom_col() + coord_flip()+
   scale_fill_manual(values=c("blue","orange"),
labels=c("Control","S4"))+ggtitle("Non-UV") +
   theme(strip.background = element_rect(fill="gray85"),
         axis.text.y.left = element_text(colour="black", size=12, face="bold"),
         axis.text.y.right = element_text(colour="black", size=12, face="bold"),
         axis.title.y = element_text(size = 12, face="bold"),
         axis.title.x = element_text(size = 12, face="bold", colour = "black"),title = element_text(size = 12, face="bold"))+
   guides(fill=guide_legend(title="Enriched Group"))
NUV_df_plotB

## plot all together
(UV_df_plotA | NUV_df_plotB) + plot_layout(guides = 'collect')+plot_annotation(tag_levels = 'A')

ggsave("Figures/FigureS8.png", width=12, height=6, dpi=400)


```

# Session info
```{r session}

session_info()

```
